### ISSUE TITLE
Menus not appearing in new windows

### ISSUE THREAD

**Issue by ulahcherubim** [wmstack/Zotero-Toggle-Bars#1]
Hi! I know this was meant to be for your personal use but since I noticed it, I hope it is okay to share what is going on.
Basically the title, menus introduced by this add-on is not appearing in new windows.
Thanks anyway for the add-on.

**Comment by wmstack** [comment]
Yeah this is a problem, as I don't know how to, or if it is even possible to add menu items to windows other than the main window. The workaround is to go to the main window and toggle the menu item, and then go back to the new window.

I'll keep it open in case someone figures out a way to add a menu item to all windows. I have tried the make-it-red plugin in Zotero, which this extension is based upon, to see if the problem is with my implementation, however, the menu item "Make It Green Instead" was not added to any new windows.

**Comment by ulahcherubim** [comment]
Since this repo has 4 stars, I just wanted to say the url in Zotero's plugins settings is not right. It goes to make-it-red, not here. This might be a problem if someone doesn't remember where they got the add-on from.
I thought a seperate issue might be overkill, but correct me if I'm wrong.

**Comment by wmstack** [comment]
Well, when I published the first release, I hadn't really changed the links of the `homepage_url` entry in the manifest from the Make It Red plugin. Anyway, I have published a new release with the latest code from Github.

Zotero is really sensitive to the contents of the `manifest.json` and having the wrong content can break it. However, the link should now be like this:
<img width="691" alt="image" src="https://github.com/user-attachments/assets/994dcf74-bb24-4b06-8e3a-3d8e8c3f2919">

I am not sure what the automatic updates url should be to track the latest Github release so I have left it unchanged.

**Comment by wmstack** [comment]
Okay update. I have done some more tweaking in the Error console and have found that I can edit the menu items of new windows, plus a way to detect when new windows have been opened:

```js
let windowListener = {
    onOpenWindow: function(aWindow) {
        // This is triggered when a new window opens
        console.log("New window opened!");

        // Optionally wait for the window to load
        let domWindow = aWindow.QueryInterface(Components.interfaces.nsIInterfaceRequestor)
                                .getInterface(Components.interfaces.nsIDOMWindow);
        domWindow.addEventListener("load", function() {
            console.log("New window has loaded.");
        }, {once: true});
    }
};

// Add the listener to detect new windows
Services.wm.addListener(windowListener);
```

You can try this out in the JS Error Console and see if it detects when you open a new window.

**Comment by wmstack** [comment]
Okay I have published the new XPI. Can you test if the menu items work across multiple windows now?

**Comment by ulahcherubim** [comment]
Now works in multiple windows without any trouble.
While updating to new version, I realized that add-on cannot be updated via Zotero's plugins tab. I think it was due to url being wrong. But this might mean other users won't get updates by you or if make-it-red updates and relases and xpi, it will be pushed.
